src/stores/chatExtras.ts

```diff
@@ -609,6 +676,127 @@ export const useChatExtrasStore = defineStore('chatExtras', () => {
 		}
 	}
 
+	/**
+	 * Converts ScheduledMessage to BigIntChatMessage format (to render in chat)
+	 *
+	 * @param token - conversation token
+	 * @param message - scheduled message object
+	 */
+	function parseScheduledToChatMessage(token: string, message: ScheduledMessage): BigIntChatMessage {
+		return {
+			token,
+			id: message.id,
+			actorId: message.actorId,
+			actorType: message.actorType,
+			actorDisplayName: actorStore.displayName,
+			message: message.message,
+			messageType: message.messageType,
+			referenceId: '',
+			systemMessage: '',
+			isReplyable: false,
+			markdown: true,
+			messageParameters: {},
+			parent: message.parent,
+			reactions: {},
+			timestamp: message.sendAt,
+			expirationTimestamp: 0,
+			threadId: message.threadId,
+			threadTitle: message.threadTitle,
+			isThread: !!message.threadId,
+			silent: message.silent,
+		}
+	}
+
+	/**
+	 * Fetch scheduled messages for given conversation
+	 *
+	 * @param token - conversation token
+	 */
+	async function fetchScheduledMessages(token: string) {
+		try {
+			const response = await getScheduledMessagesApi(token)
+			if (!scheduledMessages.value[token]) {
+				scheduledMessages.value[token] = {}
+			}
+
+			response.data.ocs.data.forEach((message) => {
+				scheduledMessages.value[token][message.id] = message
+			})
+		} catch (e) {
+			console.error('Error while fetching scheduled messages:', e)
+		}
+	}
+
+	/**
+	 * Schedule a message to be posted with given payload
+	 *
+	 * @param token - conversation token
+	 * @param payload - action payload
+	 */
+	async function scheduleMessage(token: string, payload: scheduleMessageParams) {
+		try {
+			const response = await scheduleMessageApi({ token, ...payload })
+			if (!scheduledMessages.value[token]) {
+				scheduledMessages.value[token] = {}
+			}
+			scheduledMessages.value[token][response.data.ocs.data.id] = response.data.ocs.data
+
+			await vuexStore.dispatch('setConversationProperties', {
+				token,
+				properties: {
+					hasScheduledMessages: Object.keys(scheduledMessages.value[token]).length,
+				},
+			})
```

Comment:
There is an imminent action gurantees that the store is not empty, let's not go with O(n) time complexity 
```suggestion
			scheduledMessages.value[token][response.data.ocs.data.id] = response.data.ocs.data

			await vuexStore.dispatch('setConversationProperties', {
				token,
				properties: {
					hasScheduledMessages: true,
				},
			})
```